temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowThreadExecutor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.internal.sync;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionException;
+
+public interface WorkflowThreadExecutor {
+  /**
+   * Submits a Runnable task for execution and returns a Future representing that task. The Future's
+   * {@code get} method will return {@code null} upon <em>successful</em> completion.
+   *
+   * @param task the task to submit
+   * @return a Future representing pending completion of the task
+   * @throws RejectedExecutionException if the task cannot be scheduled for execution
+   * @throws NullPointerException if the task is null
+   */
```

Comment:
Should the docstring change at all to reflect the more-specific nature of this executor?

Reply:
It's pretty much the same as the generic ThreadPool submit javadoc for #submit. But I will add an additional docstring to the class.